doc/source/data/dataset-internals.rst

```diff
@@ -139,6 +139,7 @@ Different ways of creating Datasets leads to a different starting internal forma
 * Reading tabular files (Parquet, CSV, JSON) creates Arrow blocks initially.
 * Converting from Pandas, Dask, Modin, and Mars creates Pandas blocks initially.
 * Reading NumPy files or converting from NumPy ndarrays creates Arrow blocks.
+* Reading TFRecord file creates Arrow blocks.
```

Comment:
This is the documentation update per request from https://github.com/ray-project/ray/pull/30390#issuecomment-1319339129, @stephanie-wang could you help check on behalf of ray-docs? Thanks.